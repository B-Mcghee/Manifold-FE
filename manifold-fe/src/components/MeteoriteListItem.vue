<template>
  <div>
    <base-card>
      <template v-slot:header>
        <v-toolbar-title>{{ meteorite.name }}</v-toolbar-title>
        <v-spacer></v-spacer>
        <h3>ID: {{ meteorite.id }}</h3>
      </template>
      <template v-slot:body>
        <v-container>
          <v-layout row>
            <v-flex xs12 class="pl-4">
              Mass: <span> {{ meteorite.mass }}(g)</span>
            </v-flex>
            <v-flex xs12 class="pl-4">
              Reclat:
              <span>({{ meteorite.reclat }})</span>
            </v-flex>
            <v-flex xs12 class="pl-4">
              Reclong:
              <span>({{ meteorite.reclong }})</span>
            </v-flex>
            <v-flex xs12 class="pl-4">
              Nametype:
              <span>{{ meteorite.nametype }}</span>
            </v-flex>
          </v-layout>
        </v-container>

        <!-- <v-list-item>
          <v-list-item-content>
            <div class="overline mb-3">id: {{ meteorite.id }}</div>
            <v-list-item-content class="my-2"
              >Mass: {{ meteorite.mass }} (g)</v-list-item-content
            >
            <v-list-item-subtitle
              >Geolocation:
              {{ meteorite | formatLocation }}</v-list-item-subtitle
            >
            <v-list-item-subtitle
              >Name Type: {{ meteorite.nametype }}</v-list-item-subtitle
            >
            <v-list-item-subtitle
              >Fall: {{ meteorite.fall }}</v-list-item-subtitle
            >
          </v-list-item-content>
        </v-list-item> -->
      </template>
      <template v-slot:footer>
        <v-layout>
          <v-flex xs6>
            <v-btn x-small outlined text>
              <router-link
                :to="{
                  name: 'Recclass',
                  params: { recclass: meteorite.recclass },
                }"
                >Recclass: {{ meteorite.recclass }}</router-link
              >
            </v-btn>
          </v-flex>
          <v-flex xs6>
            <v-btn x-small outlined text
              ><router-link
                :to="{
                  name: 'Year',
                  params: { year: meteorite.year },
                }"
                >Year: {{ meteorite.year | formatDate }}</router-link
              ></v-btn
            >
          </v-flex>
          <v-flex>
            <div><favorite-icon></favorite-icon></div>
          </v-flex>
          <v-flex xs6
            ><div><compare-icon></compare-icon></div>
          </v-flex>
          <v-flex xs12
            ><v-flex xs6 class="d-inline">
              <v-flex xs12> </v-flex>
              <v-flex xs12 class="d-inline "> </v-flex>
            </v-flex>
            <v-flex xs6 class="d-flex">
              <v-flex xs6 class="m-0 d-inline"></v-flex> </v-flex
          ></v-flex>
        </v-layout>
      </template>
    </base-card>
    <!-- <v-card
      raised
      color="deep-grey accent-2"
      elevation="4"
      class=" card mx-auto rounded-lg my-10"
      outlined
      max-width="250"
    >
      <v-list-item>
        <v-list-item-content>
          <div class="overline mb-3">id: {{ meteorite.id }}</div>
          <v-card-title class="headline mb-1">{{
            meteorite.name
          }}</v-card-title>
          <v-list-item-content class="my-2"
            >Mass: {{ meteorite.mass }} (g)</v-list-item-content
          >
          <v-list-item-subtitle
            >Geolocation: {{ meteorite | formatLocation }}</v-list-item-subtitle
          >
          <v-list-item-subtitle
            >Name Type: {{ meteorite.nametype }}</v-list-item-subtitle
          >
          <v-list-item-subtitle
            >Fall: {{ meteorite.fall }}</v-list-item-subtitle
          >
        </v-list-item-content>
      </v-list-item>
      <v-divider class="grey darken-6 my-3"></v-divider>
      <v-card-actions>
        <div class="d-flex flex-wrap">
          <v-btn x-small rounded outlined text class=" mx-1 my-1">
            <router-link
              :to="{
                name: 'Recclass',
                params: { recclass: meteorite.recclass },
              }"
              >Recclass: {{ meteorite.recclass }}</router-link
            >
          </v-btn>
          <v-btn x-small rounded outlined text class=" mx-1 my-1"
            ><router-link
              :to="{
                name: 'Year',
                params: { year: meteorite.year },
              }"
              >Year: {{ meteorite.year | formatDate }}</router-link
            ></v-btn
          >
          <button><favorite></favorite></button>
        </div>
      </v-card-actions>
    </v-card> -->
  </div>
</template>

<script>
import BaseCard from "./UI/BaseCard.vue";
import CompareIcon from "./UI/CompareIcon.vue";
import FavoriteIcon from "./UI/FavoriteIcon.vue";

// import Favorite from "./UI/Favorite";
export default {
  components: { BaseCard, FavoriteIcon, CompareIcon },
  props: {
    meteorite: Object,
  },
  methods: {
    resortChild(recclass) {
      console.log(recclass);
      this.$emit("resort", recclass);
    },
  },
  filters: {
    formatDate: function(value) {
      if (value) {
        var date = new Date(value);
        var parsedDate = date.getFullYear();
        return parsedDate;
      } else {
        return "N/A";
      }
    },
    formatLocation: function(values) {
      if (values) {
        var x = JSON.parse(JSON.stringify(values));
        if (x.geolocation) {
          var geolocation = JSON.parse(JSON.stringify(x.geolocation));
          return geolocation.type + " [" + geolocation.coordinates + "]";
        }
      }
    },
  },
};
</script>

<style></style>
