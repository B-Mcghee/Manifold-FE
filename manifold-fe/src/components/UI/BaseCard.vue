<template>
  <v-card max-width="250" shaped>
    <v-toolbar dense width="100%" color="#1b579e" class="white--text">
      <slot name="header"></slot>
    </v-toolbar>
    <div class="blue--text text--darken-3">
      <slot name="body"></slot>
    </div>
    <v-footer class="white--text blue-grey lighten-5">
      <slot name="footer"></slot>
    </v-footer>
  </v-card>
</template>

<script>
export default {
  components: {},
};
</script>

<style></style>
