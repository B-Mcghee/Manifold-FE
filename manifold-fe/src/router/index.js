import Vue from "vue";
import VueRouter from "vue-router";
import Home from "../views/Home.vue";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home
  },
  {
    path:"/meteorites",
    name: "Meteorite",
    component: () => import(/* webpackChunkName: "Meteorite" */'../views/Meteorite/Meteorite.vue')
  },
  {
    path:"/favorites",
    name: "Favorite",
    component: () => import('../components/layouts/FavoriteList.vue')
  },
  {
    path:"/comparison",
    name: "Compare",
    component: () => import('../components/layouts/CompareList.vue')
  },
  {
    path: "/recclass/:recclass",
    name: "Recclass",
    component: () => import('../components/layouts/Recclass.vue'),
  },
  {
    path: "/year/:year?",
    name: "Year",
    component: () => import('../components/layouts/Year.vue'),
  },

  // {
  //   path: "/about",
  //   name: "About",
  //   // route level code-splitting
  //   // this generates a separate chunk (about.[hash].js) for this route
  //   // which is lazy-loaded when the route is visited.
  //   component: () =>
  //     import(/* webpackChunkName: "about" */ "../views/About.vue")
  // }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

export default router;
