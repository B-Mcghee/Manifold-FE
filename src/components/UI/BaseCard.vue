<template>
  <v-card raised max-width="250" shaped>
    <v-toolbar dense width="100%" color="#1b579e" class="white--text">
      <slot name="header"></slot>
    </v-toolbar>
    <div class="grey--text text--darken-5 my-4">
      <slot name="body"></slot>
    </div>
    <v-footer class="white--text blue-grey lighten-5 pb-3">
      <slot name="footer"></slot>
    </v-footer>
  </v-card>
</template>

<script>
export default {
  name: "base-card",
  components: {}
};
</script>

<style></style>
