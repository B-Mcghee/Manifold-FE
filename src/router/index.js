import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);

const routes = [
  {
    path: "/meteorites",
    name: "Meteorite",
    component: () =>
      import(
        /* webpackChunkName: "Meteorite" */ "../views/Meteorite/Meteorite.vue"
      ),
    alias: "/"
  },
  {
    path: "/meteorites/category/:category",
    name: "Category",
    component: () => import("@/views/Meteorite/Meteorite.vue")
  },

  {
    path: "/meteorites/properties/:property/value/:value",
    name: "Property",
    component: () => import("@/views/Meteorite/Meteorite.vue")
  },
  { path: "/404", component: () => import("@/views/Error") },
  { path: "*", redirect: "/404" }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

export default router;
