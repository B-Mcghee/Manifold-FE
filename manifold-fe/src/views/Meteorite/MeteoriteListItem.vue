<template>
  <div>
    <base-card class="mx-auto my-2" data-testid="base-card">
      <template v-slot:header>
        <v-tooltip v-if="meteorite.name.length > 13" top>
          <template v-slot:activator="{ on, attrs }">
            <v-toolbar-title v-bind="attrs" v-on="on" data-testid="toolbar">{{
              meteorite.name | formatName
            }}</v-toolbar-title>
          </template>
          <span>{{ meteorite.name }}</span>
        </v-tooltip>
        <v-toolbar-title v-else>{{
          meteorite.name | formatName
        }}</v-toolbar-title>

        <v-spacer></v-spacer>
        <h3 class="text-h6">ID: {{ meteorite.id }}</h3>
      </template>
      <template v-slot:body>
        <v-container>
          <v-layout row>
            <v-flex xs12 class="pl-4 ">
              Mass:
              <span class="blue--text text--darken-3">
                {{ meteorite.mass }}(g)</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclat:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclat }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclong:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclong }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Nametype:
              <span class="blue--text text--darken-3">{{
                meteorite.nametype
              }}</span>
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Fall:
              <span class="blue--text text--darken-3">{{
                meteorite.fall
              }}</span>
            </v-flex>
          </v-layout>
        </v-container>
      </template>
      <template v-slot:footer>
        <v-layout block row class="py-3">
          <v-flex block xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1"
            >
              <router-link
                :to="{
                  name: 'Recclass',
                  params: { recclass: meteorite.recclass },
                }"
                >Reclass: {{ meteorite.recclass }}</router-link
              >
            </v-btn>
          </v-flex>
          <v-flex xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1"
              ><router-link
                :to="{
                  name: 'Year',
                  params: { year: meteorite.year },
                }"
              >
                year:{{ meteorite.year | formatDate }}</router-link
              ></v-btn
            >
          </v-flex>

          <v-flex xs6 class="mt-1 mx-auto ">
            <favorite-icon
              :favorite="JSON.parse(meteorite.id)"
              ref="icon"
            ></favorite-icon>
          </v-flex>
          <v-flex xs6 class="mt-1">
            <compare-icon :meteorite="JSON.parse(meteorite.id)"></compare-icon>
          </v-flex>
        </v-layout>
      </template>
    </base-card>
  </div>
</template>

<script>
import CompareIcon from "@/components/UI/CompareIcon.vue";
import FavoriteIcon from "@/components/UI/FavoriteIcon.vue";
import BaseCard from "@/components/UI/BaseCard.vue";
export default {
  components: { FavoriteIcon, CompareIcon, BaseCard },
  props: {
    meteorite: Object,
  },
  data: () => ({
    favorite: false,
  }),
  methods: {
    resortChild(recclass) {
      console.log(recclass);
      this.$emit("resort", recclass);
    },
    toggle() {},
  },

  filters: {
    formatDate: function(value) {
      if (value) {
        var date = new Date(value);
        var parsedDate = date.getFullYear();
        return parsedDate;
      } else {
        return "N/A";
      }
    },
    formatName: function(value) {
      if (value.length > 15) {
        return value.substring(0, 15);
      }
      return value;
    },
    formatLocation: function(values) {
      if (values) {
        var x = JSON.parse(JSON.stringify(values));
        if (x.geolocation) {
          var geolocation = JSON.parse(JSON.stringify(x.geolocation));
          return geolocation.type + " [" + geolocation.coordinates + "]";
        }
      }
    },
  },
  computed: {
    isFavorite: function() {
      return "";
    },
  },
};
</script>

<style>
a {
  text-decoration: none;
}
</style>
