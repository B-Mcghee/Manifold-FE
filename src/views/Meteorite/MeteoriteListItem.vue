<template>
  <div>
    <base-card class="mx-auto my-2" data-testid="base-card">
      <template v-slot:header>
        <v-tooltip v-if="meteorite.name.length > 13" top>
          <template v-slot:activator="{ on, attrs }">
            <v-toolbar-title v-bind="attrs" v-on="on" data-testid="toolbar">{{
              meteorite.name | formatName
            }}</v-toolbar-title>
          </template>
          <span>{{ meteorite.name }}</span>
        </v-tooltip>
        <v-toolbar-title v-else>{{
          meteorite.name | formatName
        }}</v-toolbar-title>

        <v-spacer></v-spacer>
        <h3 class="text-h6">ID: {{ meteorite.id }}</h3>
      </template>
      <template v-slot:body>
        <v-container>
          <v-layout row>
            <v-flex xs12 class="pl-4 ">
              Mass:
              <span class="blue--text text--darken-3">
                {{ meteorite.mass }}(g)</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclat:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclat }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclong:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclong }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Nametype:
              <span class="blue--text text--darken-3">{{
                meteorite.nametype
              }}</span>
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Fall:
              <span class="blue--text text--darken-3">{{
                meteorite.fall
              }}</span>
            </v-flex>
          </v-layout>
        </v-container>
      </template>
      <template v-slot:footer>
        <v-layout block row class="py-3">
          <v-flex block xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1 darkPrimary--text"
              :to="{
                name: 'Property',
                params: {
                  property: 'recclass',
                  value: meteorite.recclass,
                },
              }"
            >
              Reclass: {{ meteorite.recclass }}
            </v-btn>
          </v-flex>
          <v-flex xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1 darkPrimary--text"
              :to="{
                name: 'Property',
                params: { property: 'year', value: meteorite.year },
              }"
            >
              year:{{ meteorite.year | formatDate }}</v-btn
            >
          </v-flex>

          <v-flex xs6 class="mt-1 mx-auto ">
            <favorite-icon
              @toggleFavorite="toggleFavorite"
              :favorite="isFavorite"
              ref="icon"
            ></favorite-icon>
          </v-flex>
          <v-flex xs6 class="mt-1">
            <compare-icon
              @compareMeteorite="compareMeteorite"
              :compare="toCompare"
            ></compare-icon>
          </v-flex>
        </v-layout>
      </template>
    </base-card>
  </div>
</template>

<script>
import CompareIcon from "@/components/UI/CompareIcon.vue";
import FavoriteIcon from "@/components/UI/FavoriteIcon.vue";
import BaseCard from "@/components/UI/BaseCard.vue";
import { mapGetters, mapActions } from "vuex";

export default {
  components: { FavoriteIcon, CompareIcon, BaseCard },
  props: {
    meteorite: Object,
  },
  methods: {
    ...mapActions([
      "addToFavorites",
      "removeFavorite",
      "addComparison",
      "removeComparison",
    ]),
    checkFavorite(value) {
      return value === JSON.parse(this.meteorite.id);
    },
    toggleFavorite() {
      if (!this.favorites.includes(this.cleanId)) {
        console.log(this.addToFavorites);
        this.addToFavorites(this.cleanId);
      } else {
        this.removeFavorite(this.cleanId);
      }
    },
    compareMeteorite() {
      if (!this.meteoriteComparison.includes(this.cleanId)) {
        this.addComparison(this.cleanId);
      } else {
        this.removeComparison(this.cleanId);
      }
    },
  },

  filters: {
    formatDate: function(value) {
      if (value) {
        var date = new Date(value);
        var parsedDate = date.getFullYear();
        return parsedDate;
      } else {
        return "N/A";
      }
    },
    formatName: function(value) {
      if (value.length > 15) {
        return value.substring(0, 15);
      }
      return value;
    },
    formatLocation: function(values) {
      if (values) {
        var x = JSON.parse(JSON.stringify(values));
        if (x.geolocation) {
          var geolocation = JSON.parse(JSON.stringify(x.geolocation));
          return geolocation.type + " [" + geolocation.coordinates + "]";
        }
      }
    },
  },
  computed: {
    ...mapGetters(["favorites", "initialMeteorites", "meteoriteComparison"]),
    isFavorite() {
      const num = this.favorites.findIndex(this.checkFavorite);
      return num !== -1 ? true : false;
    },
    toCompare() {
      const num = this.meteoriteComparison.findIndex(this.checkFavorite);
      return num !== -1 ? true : false;
    },
    cleanId: function() {
      return JSON.parse(this.meteorite.id);
    },
  },
};
</script>

<style>
a {
  text-decoration: none;
}
</style>
