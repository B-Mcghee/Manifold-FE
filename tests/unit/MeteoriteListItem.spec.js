import { createLocalVue, shallowMount, mount } from "@vue/test-utils";
import Vuetify from "vuetify";
import BaseCard from "@/components/UI/BaseCard.vue";
import MeteoriteListItem from "@/views/Meteorite/MeteoriteListItem.vue";
import Vuex from 'vuex'
import router from 'vue-router'
import VueRouter from "vue-router";
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(router)

describe("MeteoriteListItem.vue", () => {
  let wrapper;
  let vuetify;
  let store;
let getters;
let state;
let router;
let meteorite;
  beforeEach(() => {
     meteorite = {
        name: "Aachen",
        id: "1",
        nametype: "Valid",
        recclass: "L5",
        mass: "21",
        fall: "Fell",
        year: "1880-01-01T00:00:00.000",
        reclat: "50.775000",
        reclong: "6.083330",
        geolocation: {
          type: "Point",
          coordinates: [6.08333, 50.775]
        }
      }

      getters = {
            favorites: (state) => state.favoriteMeteorites,
         initialMeteorites: (state) => state.meteorites,
         meteoriteComparison: (state) => state.meteoriteComparison
      }
      state = {
          meteorites:[
            {
                name: "Aachen",
                id: "1",
                nametype: "Valid",
                recclass: "L5",
                mass: "21",
                fall: "Fell",
                year: "1880-01-01T00:00:00.000",
                reclat: "50.775000",
                reclong: "6.083330",
                geolocation: {
                  type: "Point",
                  coordinates: [6.08333, 50.775]
                }
              },
              {
                name: "Abee",
                id: "6",
                nametype: "Valid",
                recclass: "EH4",
                mass: "107000",
                fall: "Fell",
                year: "1952-01-01T00:00:00.000",
                reclat: "54.216670",
                reclong: "-113.000000",
                geolocation: {
                  Type: "Point",
                  coordinates: [
                    -113,
                    54.21667
                  ]
                }
              }
          ],
          favoriteMeteorites:[1,2,3],
          meteoriteComparison:[2,3]
      }
    store = new Vuex.Store({
        getters,
        state
    })
    vuetify = new Vuetify();
  });


  it("renders meteorite", () => {


     wrapper = mount(MeteoriteListItem, {
      localVue,
      vuetify,
        store,
      data() {
        return {
          favorite: false
        };
      },
      propsData: {
        meteorite
      },
          stubs:['router-link', 'router-view']
    });
    expect(wrapper.html()).toMatchSnapshot();

    const title = wrapper.find(".v-toolbar__title");

    expect(title.text()).toBe(meteorite.name);
  });

  it("renders meteorite", () => {

    const $route = {
        path: "/meteorites/properties/:property/value/:value",
    name: "Property",
    }
     wrapper = mount(MeteoriteListItem, {
      localVue,
      vuetify,
        store,
      data() {
        return {
          favorite: false
        };
      },
      stubs:['router-link', 'router-view'],
      propsData: {
       meteorite
      }
     
    });

    const title = wrapper.find(".v-toolbar__title");

    expect(title.text()).toBe(meteorite.name);
  });


});
