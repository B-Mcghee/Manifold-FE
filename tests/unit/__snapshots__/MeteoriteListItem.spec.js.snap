// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeteoriteListItem.vue renders meteorite 1`] = `
<div>
  <div class="mx-auto my-2 v-card v-card--raised v-sheet v-sheet--shaped theme--light" style="max-width: 250px;" data-testid="base-card">
    <header class="white--text v-sheet theme--light v-toolbar v-toolbar--dense" style="width: 100%; height: 48px; background-color: rgb(27, 87, 158); border-color: #1b579e;">
      <div class="v-toolbar__content" style="height: 48px;">
        <div class="v-toolbar__title">Aachen</div>
        <div class="spacer"></div>
        <h3 class="text-h6">ID: 1</h3>
      </div>
    </header>
    <div class="grey--text text--darken-5 my-4">
      <div class="container">
        <div class="layout row">
          <div class="flex pl-4 xs12">
            Mass:
            <span class="blue--text text--darken-3">
              21(g)</span>
          </div>
          <div class="flex pl-4 mt-1 xs12">
            Reclat:
            <span class="blue--text text--darken-3">(50.775000)</span>
          </div>
          <div class="flex pl-4 mt-1 xs12">
            Reclong:
            <span class="blue--text text--darken-3">(6.083330)</span>
          </div>
          <div class="flex pl-4 mt-1 xs12">
            Nametype:
            <span class="blue--text text--darken-3">Valid</span>
          </div>
          <div class="flex pl-4 mt-1 xs12">
            Fall:
            <span class="blue--text text--darken-3">Fell</span>
          </div>
        </div>
      </div>
    </div>
    <footer class="v-footer white--text blue-grey lighten-5 pb-3 v-sheet theme--light">
      <div class="layout py-3 block row">
        <div class="flex block xs12">
          <router-link-stub to="[object Object]" tag="a" activeclass="v-btn--active" exactactiveclass="v-btn--active" ariacurrentvalue="page" event="click" class="text-subtitle-1 darkPrimary--text v-btn v-btn--outlined v-btn--router v-btn--text theme--light v-size--x-small" style="height: 48px; width: 100%;"><span class="v-btn__content">
            Reclass: L5
          </span></router-link-stub>
        </div>
        <div class="flex xs12">
          <router-link-stub to="[object Object]" tag="a" activeclass="v-btn--active" exactactiveclass="v-btn--active" ariacurrentvalue="page" event="click" class="text-subtitle-1 darkPrimary--text v-btn v-btn--outlined v-btn--router v-btn--text theme--light v-size--x-small" style="height: 48px; width: 100%;"><span class="v-btn__content">
            year:1880</span></router-link-stub>
        </div>
        <div class="flex mt-1 mx-auto xs6"><span class="v-tooltip v-tooltip--bottom"><!----><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="width: 100%;" id="favorite-icon" data-testid="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 111 197.29" height="2em" width="2em"><transition-stub name="bookmark"><polygon id="background" points="109.5 193.5 56.02 136.52 1.5 193.5 1.5 1.5 109.5 1.5 109.5 193.5" fill="#1b579e" stroke="#000" stroke-miterlimit="10" stroke-width="3"></polygon></transition-stub> <transition-stub name="star"><polygon id="star" points="78.31 91.9 55 80.53 31.54 91.57 36.16 67.86 17.35 50.91 43.52 47.63 55.35 26.11 66.9 47.79 93.03 51.44 74 68.13 78.31 91.9" fill="#f9ff00" stroke="#000" stroke-miterlimit="10" stroke-width="3"></polygon></transition-stub></svg></span></button></span></div>
        <div class="flex mt-1 xs6"><span class="v-tooltip v-tooltip--bottom"><!----><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="width: 100%;" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 163 158.49" height="2em" width="2em"><transition-stub name="border"><rect x="1" y="1" width="161" height="156.49" stroke="#000" stroke-miterlimit="10" stroke-width="0" fill="#fff"></rect></transition-stub> <path d="M137.45,134.74a6.9,6.9,0,0,1-4.61-2.09l-45-44c-2.28-2.23-2.92-5.46-1.38-7.06a3.28,3.28,0,0,1,2.39-.91,6.9,6.9,0,0,1,4.61,2.09l45.05,44c2.28,2.23,2.91,5.46,1.38,7.05A3.26,3.26,0,0,1,137.45,134.74Z"></path> <path d="M88.81,82.65A4.89,4.89,0,0,1,92,84.17l45.05,44c1.68,1.64,1.91,3.63,1.33,4.24a1.35,1.35,0,0,1-1,.29,4.89,4.89,0,0,1-3.21-1.51l-45-44c-1.68-1.64-1.92-3.64-1.34-4.24a1.34,1.34,0,0,1,1-.3m0-4A5.19,5.19,0,0,0,85,80.18h0c-2.33,2.41-1.69,6.83,1.42,9.87l45,44a8.82,8.82,0,0,0,6,2.65,5.21,5.21,0,0,0,3.84-1.52h0c2.32-2.41,1.68-6.83-1.43-9.87l-45-44a8.81,8.81,0,0,0-6-2.66Z"></path> <path d="M19.92,53.66c.84,50.11,74.87,50.12,75.72,0C94.8,3.54,20.77,3.54,19.92,53.66Z" fill="#c9fbff" stroke="#000" stroke-miterlimit="10" stroke-width="6"></path></svg></span></button></span></div>
      </div>
    </footer>
  </div>
</div>
`;
